storage/postgres/postgres_test.go

```diff
@@ -23,20 +23,36 @@ var store *Storage
 var userDataMock = "bar"
 
 func TestMain(m *testing.M) {
-	var err error
-	var c dockertest.ContainerID
-	c, db, err = dockertest.OpenPostgreSQLContainerConnection(15, time.Millisecond * 500)
+	c, err := dockertest.ConnectToPostgreSQL(15, time.Second, func(url string) bool {
+		var err error
+		dbTemp, err := sql.Open("postgres", url)
+		if err != nil {
+			return false
+		}
+		store = New(dbTemp)
+		if err = store.CreateSchemas(); err != nil {
+			log.Fatalf("Could not ping database: %v", err)
+		}
+
+		db = dbTemp
+
+		return db.Ping() == nil
+	})
+
 	if err != nil {
-		log.Fatalf("Could not set up PostgreSQL container: %v", err)
+		log.Fatalf("Could not connect to database: %s", err)
 	}
-	defer c.KillRemove()
 
-	store = New(db)
-	if err = store.CreateSchemas(); err != nil {
-		log.Fatalf("Could not set up schemas: %v", err)
-	}
+	retCode := m.Run()
+
+	// force teardown
+	tearDown(c)
+
+	os.Exit(retCode)
+}
 
-	os.Exit(m.Run())
```

Comment:
The defer is never called due to os.Exit() being called before
c.KillRemove() is run. Test it locally, the docker image is not removed
(not on mac at least). By actually forcing a teardown finction to be
called, we can therefore force the c.KillRemove() to be called always, in
all cases.